### DISCUSSION TITLE
How to switch from CPU to GPU?

### DISCUSSION THREAD

**Discussion by antonkiselevtdg** [awslabs/gluonts#640] [Q&A]
Hello,

Could you please help me with the following question - how to properly switch  from CPU to GPU using gluon-ts?

I have tried ctx = gpu, but it still uses just CPU. I have tried setting it multiple ways, but no result.

For e.g. I tried uninstalling mxnet and installing mxnet-cu92 instead.
no results

Environment:

Windows 10
GPU Nvidia 2080 (proper CUDA version installed)
Anaconda
gluonts 0.4.2

**Comment by StatMixedML** [comment]
You could try the following :

- make sure you have the right cuda version installed: check via *nvcc --version* in console
- pip uninstall mxnet
- pip install mxnet-cu100==1.4.1 # change -cu100 to your version
- In jupyter notebook, when using *print("Number GPU's: " + str(mx.context.num_gpus()))* you should see the number of GPUs
- in the *Trainer* argument, set *ctx = mx.context.gpu()*, assuming import mxnet as mx

**Comment by antonkiselevtdg** [comment]
Thank you for details StatMixedML.

I have tried to apply it to a different machine, but now I am getting conflict of versions:

To install CUDA I need MS VS. Only available version on MS site is 2019

I can only install CUDA 10.1 or 10.2. Cuda 10.0 doesn't want to install with VS 2019 (I have read that even some 2017 versions have this issue).

Then when I try to run:

pip uninstall mxnet
pip install*mxnet-cu101==1.4.1

I am not able to properly install mxnet-cu101 - it doesn't properly support 1.4.1 (though it should, but it doesn't work).

So now I am stuck...

I suppose I should try to get VS 2015 or early VS 2017 that works with CUDA 10 and then run
pip install*mxnet-cu100==1.4.1

OR

wait until there is mxnet/gluonts support for CUDA 10.2

Does gluon-ts work with 1.5+ version of mxnet?

**Comment by StatMixedML** [comment]
As far as I know, GluonTS currently only supports mxnet1.4.1. See also here https://github.com/awslabs/gluon-ts/issues/331.

**Comment by parimuns** [comment]
> As far as I know, GluonTS currently only supports mxnet1.4.1. See also here #331.

Hello 
 I do not have system with GPU but I am trying to run deepstate on GPU servers like Google colab but its not working.Can you please guide ,how to run there ?
